File: core/src/test/java/io/pivotal/main/InMemoryExampleModelRepository.java
package io.pivotal.main;

import java.util.ArrayList;
import java.util.List;

public class InMemoryExampleModelRepository implements ExampleModelRepository {

    private List<ExampleModel> mExampleModels = new ArrayList<>();

    @Override
    public List<ExampleModel> findAll() {
        return mExampleModels;
    }

    @Override
    public void save(ExampleModel exampleModel) {
        mExampleModels.add(exampleModel);
    }
}

File: core/src/test/java/com/billiamram/main/InMemoryExampleModelRepository.kt
package com.billiamram.main

import java.util.*

class InMemoryExampleModelRepository : ExampleModelRepository {

    private val mExampleModels = ArrayList<ExampleModel>()

    override fun findAll(): List<ExampleModel> {
        return mExampleModels
    }

    override fun save(exampleModel: ExampleModel) {
        mExampleModels.add(exampleModel)
    }
}
